File: gracejuster_assessment/javascript/models/restaurant.js
const Restaurant = (function () {
    var counter = 1
    return class Restaurant {
        constructor(name){
            this.id = counter++
            this.name = name
            this.reviews = []
            this.customers = []
            Store.data.restaurants.push(this)
        }

        all() {
            return Store.data.restaurants
        }

        find_by_name(name) {
            return Store.data.restaurants.find( 
                function(restaurant) { 
                    restaurant.name === name 
                }
            )

        }

        customers() {
            return this.customers
        }

        reviews() {
            return this.reviews 
        }

    }

}())

File: gracejuster_assessment/object_relations/app/models/restaurant.rb
class Restaurant
	attr_reader :name

	@@all = []

  def initialize(name)
  	@name = name
  	@reviews = []
  	@customers = []
  	@@all << self
  end 

  def self.all 
  	@@all
  end 

  def self.find_by_name(name)
  	self.all.find do |resturant|
  		resturant.name == name
  	end
  end 

  def customers 
  	@customers
  end 

  def reviews
  	@reviews
  end 
  
end